react/src/lib/components/SubsurfaceViewer/layers/map/mapLayer.ts

```diff
@@ -342,14 +355,32 @@ export default class MapLayer extends CompositeLayer<MapLayerProps<unknown>> {
                     const zMin = -meshZValueRange[0];
                     const xMax = xMin + xinc * xcount;
                     const yMax = yMin + yinc * ycount;
-                    const zMax = -meshZValueRange[0];
+                    const zMax = -meshZValueRange[1];
+
+                    // If map is rotated the bounding box must reflect that.
+                    const center =
+                        this.props.frame.rotPoint ?? this.props.frame.origin;
+                    const rotDeg = this.props.frame.rotDeg ?? 0;
+                    const rotRad = (rotDeg * (2.0 * Math.PI)) / 360.0;
+
+                    // Rotate x,y around "center" "rad" radians
+                    const [x0, y0] = rotate(xMin, yMin, center[0], center[1], rotRad); // eslint-disable-line
+                    const [x1, y1] = rotate(xMax, yMin, center[0], center[1], rotRad); // eslint-disable-line
+                    const [x2, y2] = rotate(xMax, yMax, center[0], center[1], rotRad); // eslint-disable-line
+                    const [x3, y3] = rotate(xMin, yMax, center[0], center[1], rotRad); // eslint-disable-line
+
+                    // Rotated bounds in x/y plane.
+                    const x_min = Math.min(x0, x1, x2, x3);
```

Comment:
Not sure why you're switching between camel and kebab case :D I think the convention is camel